tinyiiod.c

```diff
@@ -186,6 +186,33 @@ void tinyiiod_do_close(struct tinyiiod *iiod, const char *device)
 	tinyiiod_write_value(iiod, ret);
 }
 
+int32_t tinyiiod_do_writebuf(struct tinyiiod *iiod,
+			     const char *device, size_t bytes_count)
+{
+	int32_t ret;
+	size_t total_bytes = bytes_count;
+	char buf[256];
+	size_t offset = 0;
+
+	tinyiiod_write_value(iiod, bytes_count);
+	while(bytes_count) {
+		size_t bytes = bytes_count > sizeof(buf) ? sizeof(buf) : bytes_count;
+		ret = tinyiiod_read(iiod, buf, bytes);
+		if(ret > 0) {
+			ret = iiod->ops->write_data(device, buf, offset, ret);
+			offset += ret;
+			if (ret < 0)
+				return ret;
+			bytes_count -= ret;
+		} else
+			return ret;
+	}
+	ret = iiod->ops->store_data(device, total_bytes);
```

Comment:
does this store in memory ? and write_value to device ?

Reply:
yes